### PR TITLE
Copy PR template from main project

### PR DESCRIPTION
**Pull request by dbwiddis**
Signed-off-by: Daniel Widdis <widdis@gmail.com>

### Description
This PR copies over the PR template from the main OpenSearch project site, which includes a check list prompting the submitter to write tests and add javadocs, and prompts reviewers to check for the same.

### Issues Resolved
Part of a strategy to address #27.  Additional checks will be added in a later PR.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ 
+### Issues Resolved
+[List any issues this PR will resolve]
+ 
+### Check List
+- [ ] New functionality includes testing.
+  - [ ] All tests pass
+- [ ] New functionality has been documented.
+  - [ ] New functionality has javadoc added
+- [ ] Commits are signed per the DCO using --signoff
```
Comment (peternied) [inline .github/pull_request_template.md:12]: I don't think we should include this line item; its explicitly enforced by the PR DCO check enabled on all opensearch-project repositories and the statement at the bottom

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+### Description
+[Describe what this change achieves]
+ 
+### Issues Resolved
+[List any issues this PR will resolve]
+ 
+### Check List
+- [ ] New functionality includes testing.
+  - [ ] All tests pass
```
Comment (peternied) [inline .github/pull_request_template.md:9]: Shouldn't this enforced by the PR check, can we remove this line if its already checked?

**Review by peternied — CHANGES_REQUESTED** [review]
Doesn't look like this follows the template from [opensearch-project/.github](https://github.com/opensearch-project/.github/blob/main/PULL_REQUEST_TEMPLATE.md) is cleaner.  What do you think about using it instead?

**Comment by dbwiddis** [comment]
> Doesn't look like this follows the template from [opensearch-project/.github](https://github.com/opensearch-project/.github/blob/main/PULL_REQUEST_TEMPLATE.md) is cleaner. What do you think about using it instead?

It follows that template and adds the checklist to it. It's currently active for PRs on OpenSearch site (where I copied it from and it takes priority over the organization-wide default):
https://github.com/opensearch-project/OpenSearch/blob/main/.github/pull_request_template.md

**Comment by owaiskazi19** [comment]
> OK, so this is probably a bigger question/comment/concern.
> 
> We should have a standardized template across all `opensearch-project` repos. The template @dblock has here accomplishes this: https://github.com/opensearch-project/.github/blob/main/PULL_REQUEST_TEMPLATE.md

+1 to this.

> I don't think it's a good idea to have differing templates across the projects. @peternied raises some good points about whether we need to include items that are already part of the automated checks (tests pass; DCO signoff) and I'll be adding another PR to include javadoc checks as part of the gradle build.

Regarding the checklist
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass.  -> This is covered by gradle check. We can remove this for OpenSearch and opensearch-sdk.
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added. -> Now since OpenSearch has strict javadoc rule as a part of gradle check. We can remove this as well for OpenSearch repo. SDK repo doesn't have the rule integrated yet.
- [ ] Commits are signed per the DCO using --signoff -> We have DCO bot for all the repos of opensearch-project. We can remove this too.

**Comment by saratvemulapalli** [comment]
+1 to @peternied. Lets just use whats the standard for our project. 
@dbwiddis there are great questions here like redundant DCO, test pass etc. Probably a PR to `.github` upstream?

**Pull request closed without merging**